python/pyspark/ml/util.py

```diff
@@ -433,17 +433,16 @@ def _get_metadata_to_save(instance, sc, extraMetadata=None, paramMap=None):
 
         # User-supplied param values
         params = instance._paramMap
-        jsonParams = {}
+
+        jsonParams={}
         if paramMap is not None:
             jsonParams = paramMap
         else:
             for p in params:
                 jsonParams[p.name] = params[p]
 
         # Default param values
-        jsonDefaultParams = {}
-        for p in instance._defaultParamMap:
-            jsonDefaultParams[p.name] = instance._defaultParamMap[p]
+        jsonDefaultParams={p.name:instance._defaultParamMap[p] for p in instance._defaultParamMap}
```

Comment:
This is more a personal preference. I don't see any gain by this, and don't think we should fix these.